enterprise/dbcrypt/doc.go

```diff
@@ -0,0 +1,34 @@
+// Package dbcrypt provides a database.Store wrapper that encrypts/decrypts
+// values stored at rest in the database.
+//
+// Encryption is done using Ciphers, which is an abstraction over a set of
+// encryption keys. Each key has a unique identifier, which is used to
+// uniquely identify the key whilst maintaining secrecy.
+//
+// Currently, AES-256-GCM is the only implemented cipher mode.
+// The Cipher is currently used to encrypt/decrypt the following fields:
+// - database.UserLink.OAuthAccessToken
+// - database.UserLink.OAuthRefreshToken
+// - database.GitAuthLink.OAuthAccessToken
+// - database.GitAuthLink.OAuthRefreshToken
+// - database.DBCryptSentinelValue
+//
+// Multiple ciphers can be provided to support key rotation. The primary cipher
+// is used to encrypt and decrypt all data. Secondary ciphers are only used
+// for decryption and, as a general rule, should only be active when rotating
+// keys.
+//
+// Encryption keys are stored in the database in the table `dbcrypt_keys`.
+// The table has the following schema:
+//   - number: the key number. This is used to avoid conflicts when rotating keys.
+//   - created_at: the time the key was created.
+//   - active_key_digest: the SHA256 digest of the active key. If null, the key has been revoked.
+//   - revoked_key_digest: the SHA256 digest of the revoked key. If null, the key has not been revoked.
+//   - revoked_at: the time the key was revoked. If null, the key has not been revoked.
+//   - test: the encrypted value of the string "coder". This is used to ensure that the key is valid.
```

Comment:
nit: does it make sense to keep it? In theory, somebody can alter the "key and" "test" with values from another record. Key will be valid, and "test" will be decrypted too.

FYI I'm just suggesting that if this mechanism isn't bulletproof, let's simplify/remove it.

Reply:
At startup, we validate that we have all required encryption keys to access the database. Part of this includes being able to decrypt the 'test' value. I'd rather this happen immediately at startup than at some undefined later point in time.